### PR TITLE
Shrilekha

### PR DESCRIPTION
**Pull request by Shrilekha68**
Again changes made.

### CODE REVIEW

File: src/app/employee-details/bar-chart/bar-chart.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Comment (aarthi1246) [inline src/app/employee-details/bar-chart/bar-chart.component.spec.ts:2]: remove extra space

File: src/app/employee-details/bar-chart/bar-chart.component.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+})
+export class BarChartComponent {
+  chartdata: any = [];
+  constructor(private service: DataService) {}
+  ngOnInit() {
+    this.datas();
+  }
+  datas() {
+    this.service.getdata().subscribe((res: any) => {
+      this.chartdata = res.map((resp: any) => resp.deportment);
```
Comment (aarthi1246) [inline src/app/employee-details/bar-chart/bar-chart.component.ts]: naming convention

**Review by aarthi1246 — APPROVED** [review]

**Review by HarishDass — APPROVED** [review]

File: src/app/employee-details/servicess/data.service.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { Data } from 'src/interface/data';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DataService {
+  constructor(private http: HttpClient) {}
+  getdata(): Observable<Data[]> {
+    return this.http.get<Data[]>('http://localhost:3000/Employeedetails');
+  }
```
Comment (bharathidhasanm2s) [inline src/app/employee-details/servicess/data.service.ts]: use environment format is 'localhost:3000'

File: src/app/employee-details/bar-chart/bar-chart.component.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Chart.register(...registerables);
+import { DataService } from '../servicess/data.service';
+
+@Component({
+  selector: 'app-bar-chart',
+  templateUrl: './bar-chart.component.html',
+  styleUrls: ['./bar-chart.component.css'],
+})
+export class BarChartComponent {
+  chartdata: any = [];
```
Comment (bharathidhasanm2s) [inline src/app/employee-details/bar-chart/bar-chart.component.ts]: remove any

File: package-lock.json [hunk 5]
```diff
@@ -17,7 +17,10 @@
         "@angular/platform-browser": "^16.2.0",
```
Comment (bharathidhasanm2s) [inline package-lock.json:17]: Don't change the package -lock.json

File: db.json [hunk 6]
```diff
@@ -0,0 +1,363 @@
+{"Employeedetails" :
```
Comment (bharathidhasanm2s) [inline db.json]: Remove Unwanted space

File: package.json [hunk 7]
```diff
@@ -19,7 +19,10 @@
     "@angular/platform-browser": "^16.2.0",
     "@angular/platform-browser-dynamic": "^16.2.0",
     "@angular/router": "^16.2.0",
+    "bootstrap": "^5.3.1",
```
Comment (bharathidhasan2002) [inline package.json:22]: remove the G and jsn

File: src/app/employee-details/employee-details.module.ts [hunk 8]
```diff
@@ -4,9 +4,10 @@ import { CommonModule } from '@angular/common';
 import { EmployeeDetailsRoutingModule } from './employee-details-routing.module';
 import { EmployeeDetailsComponent } from './employee-details.component';
 import { SharedModuleModule } from '../shared-module/shared-module.module';
+import { BarChartComponent } from './bar-chart/bar-chart.component';
```
Comment (bharathidhasan2002) [inline src/app/employee-details/employee-details.module.ts]: change the BarChart to Employee-Dashboard

File: src/app/employee-details/employee-details.component.html [hunk 9]
```diff
@@ -1 +1 @@
-<p>Employee Working</p>
+<app-bar-chart></app-bar-chart>
```
Comment (bharathidhasan2002) [inline src/app/employee-details/employee-details.component.html]: Use the Route Method

**Review by aarthi1246 — APPROVED** [review]

**Review by HarishDass — APPROVED** [review]

**Pull request merged**